src/common/components/messages/ProfileMessages.tsx

```diff
@@ -0,0 +1,385 @@
+          messagesInConversation.every((m) => m.sender.toBase58() === publicKey) &&
+          recipient ? (
+            <div className="mx-auto max-w-md rounded-full p-4 text-sm shadow-lg shadow-black">
+              Until the other sender responds, your messages will only be available in this browser.
+            </div>
+          ) : null}
+        </div>
+
+        <form
+          className="mt-full flex items-center justify-between space-x-4 border-t border-gray-800 p-5 pb-10"
+          onSubmit={sendMessage}
+        >
+          {newMessageText.length < 100 ? (
+            <input
+              name="message"
+              ref={newMessageInput as RefObject<HTMLInputElement>}
+              disabled={messageTransactionInProgress || !recipient?.address}
+              className="w-full resize-none rounded-lg  bg-gray-900 px-4 py-2 text-white ring-1  ring-gray-800  focus:ring-white disabled:bg-gray-800 "
+              placeholder={'Message ' + shortenAddress(recipient?.address || '') + '...'}
+              value={newMessageText}
+              autoFocus={true}
+              onChange={(e) => setNewMessageText(e.target.value)}
+            />
+          ) : (
+            <textarea
+              name="message"
+              ref={newMessageInput as RefObject<HTMLTextAreaElement>}
+              disabled={messageTransactionInProgress || !recipient?.address}
+              className="w-full resize-none rounded-lg  bg-gray-900 px-4 py-2 text-white ring-1  ring-gray-800  focus:ring-white"
+              placeholder={'Message ' + shortenAddress(recipient?.address || '') + '...'}
+              value={newMessageText}
+              autoFocus={true}
+              onChange={(e) => setNewMessageText(e.target.value)}
+              onFocus={function (e) {
+                // used to keep focus when the input changes to a text area
+                var val = e.target.value;
+                e.target.value = '';
+                e.target.value = val;
+              }}
+            />
+          )}
+
+          <div className="top-2 right-2">
+            <Button5 loading={messageTransactionInProgress} v="secondary" type="submit">
+              Send
+            </Button5>
+          </div>
+        </form>
+      </div>
+    </div>
+  );
+};
+
+function Contact(props: { selected: boolean; onSelect?: () => void; user: User }): JSX.Element {
+  return (
+    <div
+      id={props.user.address}
+      className={classNames(
+        ' flex cursor-pointer items-center space-x-2 rounded-md  px-2 py-2',
+        props.selected && 'bg-gray-800'
+      )}
+      onClick={props.onSelect}
+    >
+      <ProfilePFP user={props.user} />
+
+      <div className="text-base">
+        {props.user.profile?.handle ?? shortenAddress(props.user.address)}
+      </div>
+    </div>
+  );
+}
+
+const ONE_WEEK_IN_MS = 7 * 24 * 3600 * 1000;
+
+function MessageBlock(props: { myPubkey: string; messageBlock: MessageAccount[] }) {
```

Comment:
I notice you have a return type on `Contact` but not here. Would love to have it everywhere you can